lambda_functions/va_profile/va_profile_opt_in_out_lambda.py

```diff
@@ -365,6 +395,42 @@ def va_profile_opt_in_out_lambda_handler(  # noqa: C901
                     }
                 )
 
+        va_profile_id = bio['vaProfileId']
+
+        # Send Comp and Pen Opt-In confirmation if anticipated status code still 200
+        # And if Opt-In confirmation (bio['allowed'] == True)
+        if post_response['statusCode'] == 200 and bio['allowed']:
+            response = send_comp_and_pen_opt_in_confirmation(va_profile_id)
+
+            # Save notification_id from POST sms response if method returned
+            # a value AND the response status code is 201.
+            if response is None:
+                logger.critical(
+                    'Could not send Comp and Pen opt-in confirmation to VAProfileId: %s. No response status or response body to record.',
+                    va_profile_id,
+                )
+            elif response.status != 201:
+                response_data = response.read().decode()
+                response_json = json.loads(response_data)
+                logger.critical(
+                    'Could not send Comp and Pen opt-in confirmation to VAProfileId: %s. Response status: %s, Response: %s',
+                    va_profile_id,
+                    response.status,
+                    response_data,
+                )
+            else:
+                response_data = response.read().decode()
+                response_json = json.loads(response_data)
+                notification_id = response_json['id']
+
+                logger.info(
+                    'Sent Comp and Pen opt-in confirmation to VAProfileId: %s with notification_id: %s. Response status: %s',
+                    va_profile_id,
+                    notification_id,
+                    response.status,
+                )
+                save_notification_id_to_cache(va_profile_id, notification_id, bio['sourceDate'])
```

Comment:
I don't believe this needs to be in the finally block.

We need to protect our integration with VA Profile by wrapping this to prevent exceptions from bubbling up. Given the code is likely being rewritten I'm okay wrapping the whole thing in a try/except block.

Reply:
Good call. Updated.